### PR TITLE
Feature/issue#39 migrate site to next js

### PR DESCRIPTION
**Pull request by notjustinshaw**
Take your time with this one. 

I ~~don't feel too comfortable here yet~~ ready for review. 

~~I think I broke the `/discord` endpoint~~ (fixed)

### CODE REVIEW

**Comment by notjustinshaw** [comment]
This is a work in progress (WIP). Do not merge this until we fix the bugs that we have noted and all the approvers have seen the update.

File: package.json
Comment (notjustinshaw) [inline package.json]: this should be blubbr

File: package.json
Comment (notjustinshaw) [inline package.json:4]: we should keep the homepage

File: public/vercel.svg
Comment (notjustinshaw) [inline public/vercel.svg]: we need to delete this

File: src/App.test.tsx
Comment (notjustinshaw) [inline src/App.test.tsx:1]: we should keep a test file

File: utils/ScrollReveal.tsx
Comment (notjustinshaw) [inline utils/ScrollReveal.tsx:30]: I am not sure if this is the right way to go about doing this.

File: components/layout/partials/Logo.tsx
Comment (notjustinshaw) [inline components/layout/partials/Logo.tsx:18]: This <Image/> should be the next/image component rather than our own custom Image component. If we use their react component, we get automatic image resizing and compression and other optimizations for server-speed.

File: .vscode/settings.json
Comment (notjustinshaw) [inline .vscode/settings.json]: Let's remove the diffs on this file (I believe they are the same except for whitespace differences).

File: .gitignore
Comment (notjustinshaw) [inline .gitignore:1]: I need to go through here and figure out what we need to change and keep and update.

File: .github/actions/build/action.yaml
Comment (notjustinshaw) [inline .github/actions/build/action.yaml]: keep this

File: .github/actions/deploy/action.yaml
Comment (notjustinshaw) [inline .github/actions/deploy/action.yaml:1]: Keep this

File: .github/actions/lint/action.yaml
Comment (notjustinshaw) [inline .github/actions/lint/action.yaml]: keep this

File: .github/actions/test/action.yaml
Comment (notjustinshaw) [inline .github/actions/test/action.yaml]: keep this

File: .github/workflows/main.yaml
Comment (notjustinshaw) [inline .github/workflows/main.yaml]: keep this

**Review by notjustinshaw — COMMENTED** [review]
Left 13 to-do's before we merge this guy

File: package.json
Comment (notjustinshaw) [reply package.json]: done.

File: package.json
Comment (notjustinshaw) [reply package.json:4]: done.

File: public/vercel.svg
Comment (notjustinshaw) [reply public/vercel.svg]: done

File: src/App.test.tsx
Comment (notjustinshaw) [reply src/App.test.tsx:1]: I'm gonna open a separate issue for this one (see #30)

File: .vscode/settings.json
Comment (notjustinshaw) [reply .vscode/settings.json]: done

File: components/layout/partials/Logo.tsx
Comment (notjustinshaw) [reply components/layout/partials/Logo.tsx:18]: Will create a new issue for this one (see #42)

File: utils/ScrollReveal.tsx
Comment (notjustinshaw) [reply utils/ScrollReveal.tsx:30]: I'll make an issue for this one as well (see #43)

**Review by sorenrood — APPROVED** [review]
**LGTM**
penguin_dance

**Pull request merged**
